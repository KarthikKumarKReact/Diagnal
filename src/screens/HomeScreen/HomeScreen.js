import { useEffect, useState } from 'react'
import CardList from '../CardList/CardList'
import { getData } from '../../services/data'

const HomeScreen = () => {
    const [selectedGenre, setSelectedGenre] = useState('Romantic Comedy')
    const [movieList, setMovieList] = useState([])
    const [pagination, setPagination] = useState({ currPage: 0, totalItems: 0 })

    useEffect(() => {
        getList()
    }, [])

    const getList = async () => {
        const { currPage, totalItems } = pagination
        if ((totalItems === movieList.length) && currPage) {
            return
        }
        const data = await getData(currPage)
        const newData = data.page['content-items'].content
        const newCurrPage = parseInt(data.page['page-num-requested'])
        const newTotalItems = parseInt(data.page['total-content-items'])
        setPagination({ currPage: newCurrPage, totalItems: newTotalItems })
        setMovieList((movieList) => [...movieList, ...newData])
    }
    return (
        <>
            {selectedGenre ?
                <CardList title={selectedGenre} movies={movieList} getList={getList} />
                : null
            }
        </>
    )
}

export default HomeScreen