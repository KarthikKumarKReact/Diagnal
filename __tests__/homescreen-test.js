import 'react-native';
import React from 'react';
import HomeScreen from '../src/screens/HomeScreen/HomeScreen'
import renderer from 'react-test-renderer';
import { render, waitFor } from '@testing-library/react-native';

it('Connection UI Snapshot', () => {
    const tree = renderer.create(<HomeScreen />).toJSON();
    expect(tree).toMatchSnapshot();
});

test('renders without error', () => {
    render(<HomeScreen />);
  });
