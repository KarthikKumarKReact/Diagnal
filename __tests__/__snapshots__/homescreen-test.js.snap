// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Connection UI Snapshot 1`] = `
<View
  style={
    {
      "backgroundColor": "#000000",
      "flex": 1,
    }
  }
>
  <View
    style={
      {
        "alignItems": "center",
        "flexDirection": "row",
        "height": 50,
        "justifyContent": "space-between",
        "marginHorizontal": 10,
      }
    }
  >
    <View
      style={
        {
          "alignItems": "center",
          "flexDirection": "row",
        }
      }
    >
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "opacity": 1,
          }
        }
      >
        <Image
          source={NaN}
          style={
            {
              "height": 20,
              "tintColor": "#ffffff",
              "width": 20,
            }
          }
        />
      </View>
      <Text
        style={
          {
            "color": "#ffffff99",
            "fontSize": 18,
            "marginLeft": 20,
          }
        }
      >
        Romantic Comedy
      </Text>
    </View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "marginRight": 5,
          "opacity": 1,
        }
      }
    >
      <Image
        source="test-file-stub"
        style={
          {
            "height": 20,
            "tintColor": "#ffffff",
            "width": 20,
          }
        }
      />
    </View>
  </View>
  <RCTScrollView
    ListEmptyComponent={<NothingFound />}
    contentContainerStyle={
      {
        "justifyContent": "space-between",
      }
    }
    data={[]}
    getItem={[Function]}
    getItemCount={[Function]}
    keyExtractor={[Function]}
    onContentSizeChange={[Function]}
    onEndReached={[Function]}
    onEndReachedThreshold={0.5}
    onLayout={[Function]}
    onMomentumScrollBegin={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    showsVerticalScrollIndicator={false}
    stickyHeaderIndices={[]}
    style={
      {
        "backgroundColor": "#050505",
        "marginHorizontal": 6,
      }
    }
    viewabilityConfigCallbackPairs={[]}
  >
    <View>
      <View
        style={
          {
            "alignItems": "center",
            "marginTop": 70,
          }
        }
      >
        <Text
          style={
            [
              {
                "color": "#ffffff",
              },
              {
                "fontSize": 24,
                "fontWeight": "bold",
              },
            ]
          }
        >
          No Results Found
        </Text>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
